### DISCUSSION TITLE
Gratitude

### DISCUSSION THREAD

**Discussion by Faerre** [joqueka/bf4py#9] [General]
Hello,
Just want to say thank you for extremly usefull library you`ve made.
I use it on daily basis to collect more details about derivatives (knock-out certificates).
Fortunatly, libarary helps to close a gap of inforamtion which I cannot even get with any super-fancy APIs.

Please continue to develope it.

P.S. I think it worth if you write some details how to buy you a coffee

**Comment by joqueka** [comment]
Hi,
I'm very happy to hear this! Currently I'm working on putting it on a object-oriented base and this is good motivation :)

More than a coffee I would like to know how you use the collected data and which information you're looking for. I started this project for the same reason and to get more understanding of the financial markets. You can send me a DM if you like.

Cheers

**Reply by Faerre** [reply]
Hi joqueka,
Sorry for late reply.
Mainly my task is to automate a trading with knock-out certificates, I`ve found that Interactive broker API is the most effcient tool for that.
Howerver there is a lack of functionality - I cannot get a precise name or type of certificate, as result I can generate only a list with knock-out certificates but also factor-certificates that I cannot filter out in advance.
Sad thing that API cannot return the name of certificate where would be clear what is that knock-out or factor.

So from that moment your code is coming and gets that information from boersa directly. Later on I can filter out factor cerificates just by their market names.
As a bonus I can pull the issuers of certificates as far as IB API cannot do it as well

**Reply by joqueka** [reply]
Wow I didn't expected my package to be used in productive environments, that's pretty cool!

For that kind of task you should also have a look at the detailed derivatives information, maybe you can also use things like EUSIPA codes or absence of knock-out barriers for filtering. 

boerse-frankfurt also has a search functionality but I haven't implemented it yet because there are lots of search parameters. Maybe I'll have a look at this next and then it might be as useful as the IB API.
